dbms/src/Flash/ResourceControl/LocalAdmissionController.cpp

```diff
@@ -13,19 +13,491 @@
 // limitations under the License.
 
 #include <Flash/ResourceControl/LocalAdmissionController.h>
+#include <etcd/rpc.pb.h>
+#include <pingcap/kv/Cluster.h>
 
 namespace DB
 {
-std::unique_ptr<MockLocalAdmissionController> LocalAdmissionController::global_instance = nullptr;
+ResourceGroupPtr LocalAdmissionController::getOrFetchResourceGroup(const std::string & name)
+{
+    ResourceGroupPtr group = findResourceGroup(name);
+    if (group != nullptr)
+        return group;
+
+    resource_manager::GetResourceGroupRequest req;
+    req.set_resource_group_name(name);
+    resource_manager::GetResourceGroupResponse resp;
+
+    try
+    {
+        resp = cluster->pd_client->getResourceGroup(req);
+    }
+    catch (...)
+    {
+        auto err_msg = getCurrentExceptionMessage(false);
+        throw ::DB::Exception(fmt::format("getOrFetchResourceGroup({}) failed: {}", name, err_msg));
+    }
+
+    RUNTIME_CHECK_MSG(!resp.has_error(), "getOrFetchResourceGroup({}) failed: {}", name, resp.error().message());
+
+    std::string err_msg = isGACRespValid(resp.group());
+    RUNTIME_CHECK_MSG(err_msg.empty(), "getOrFetchResourceGroup({}) failed: {}", name, err_msg);
+
+    return addResourceGroup(resp.group()).first;
+}
+
+void LocalAdmissionController::startBackgroudJob()
+{
+    try
+    {
+        unique_client_id = etcd_client->acquireServerIDFromPD();
+    }
+    catch (...)
+    {
+        auto err_msg = getCurrentExceptionMessage(false);
+        LOG_FATAL(log, err_msg);
```

Comment:
Need to crash tiflash?
What happens if tiflash continues to run?

Reply:
We must have correct `unique_client_id` for GAC(GAC use this id to distinguish different LAC and balance token).

Reply:
Or maybe we can deadloop retry(sleep X second before next retry) and LOG_ERROR when got exception?

Reply:
So it looks like `std::terminate()` need to call here

Reply:
ok. I thought `LOG_FATAL` would automatically crash for  looks I'm wrong.